introduction.Rmd

```diff
@@ -184,6 +184,309 @@ We call the combination of these two steps the __quote and unquote__ pattern. Th
 Most of the time though, we don't need to create blueprints manually. We'll get them by quoting the arguments supplied by users. This gives your functions the same usage and feel as tidyverse verbs.
 
 
+# Do you need tidy eval? {#sec:do-you-need}
+
+In the field of computer science, frameworks like tidy evaluation are known as metaprogramming. Modifying the blueprints of computations amounts to programming the program, i.e. metaprogramming. In other languages, this type of approach is often seen as a last resort because it requires new skills and might make your code harder to read. Things are different in R because of the importance of data masking functions, but it is still good advice to consider other options before turning to tidy evaluation. In this section, we review several strategies for solving programming problems with tidyverse packages.
+
+Before diving into tidy eval, make sure to know about the fundamentals of programming with the tidyverse:
+
+* [Fixed column names](#sec:fixed-colnames). A solid function taking data frames with fixed column names is better than a brittle function that uses tidy eval.
+
+* [Automating loops](#sec:automating-loops). dplyr excels at automating loops. Acquiring a good command of rowwise vectorisation and columnwise mapping may prove very useful.
+
+Tidy evaluation does not have to be hard. It features several lightweight techniques that you can easily pick up in your workflow:
+
+* Pass expressions through `{{` and `...`.
+* Pass column names to `.data[[` and `one_of()`.
+
+All these techniques make it possible and even easy to reuse existing components of tidyverse grammars and compose them into new functions.
+
+
+## Fixed column names {#sec:fixed-colnames}
+
+A simple solution is to write functions that expect data frames containing specific column names. If the computation always operates on the same columns and nothing varies, you don't need any tidy eval. On the other hand, your users must ensure the existence of these columns as part of their data cleaning process.
```

Comment:
Perhaps even be explicit that this works well if you're writing functions primarily for your own use, or maybe an example scenario such as for use with a specific API (I do this all the time with NBA data).

Reply:
Perhaps:

This is why this technique primarily makes sense when you're writing functions tailored to your own data analysis uses, or perhaps in functions that interface with a specific web API for retrieving data. In general, fixed column names are task specific.

Reply:
That sounds great!